src/DistributedLocking/Consul/LockProvider.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Cascade\DistributedLocking\Consul;
+
+use Cascade\DistributedLocking\ILockProvider;
+use Cascade\DistributedLocking\ILockSessionProvider;
+use SensioLabs\Consul\Services\KV;
+
+class LockProvider implements ILockProvider
+{
+    /** @var KV */
+    private $consulKeyValue;
+
+    /** @var ILockSessionProvider */
+    private $sessionProvider;
+
+    /**
+     * @param ILockSessionProvider $sessionProvider
+     * @param KV $consulKeyValue
+     */
+    public function __construct(ILockSessionProvider $sessionProvider, KV $consulKeyValue)
```

Comment:
I would be OK with `$kvClient` -- I think `$kv` (and `KV`) is actually sort of too short a name for anything :-b